### PR TITLE
Add more context flags and getters to `MeasuredTemplateDocumentPF2e`

### PR DESCRIPTION
**Pull request by In3luki**

### CODE REVIEW

**Comment by stwlam** [comment]
Can `MeasuredTemplateDocument` be constructed with a flag such that it knows what kind of pathfinder area it is?

**Comment by In3luki** [comment]
I have removed the `drawPreview` arguments and moved the new context to `MeasuredTemplateDocumentPF2e#flags`.
I've also created some new getters on the document and the template object for easier access of the flags.

File: src/module/canvas/measured-template.ts [hunk 1]
```diff
@@ -48,18 +66,17 @@ class MeasuredTemplatePF2e<
...
 
-        canvas.stage.on("mousemove", this.#onPreviewMouseMove);
-        canvas.stage.on("mousedown", this.#onPreviewLeftClick);
-        canvas.stage.on("rightdown", this.#onPreviewRightClick);
-        canvas.app.view.addEventListener?.("wheel", this.#onPreviewMouseWheel, this.#wheelListenerOptions);
-
         if (canvas.grid.type === CONST.GRID_TYPES.SQUARE) {
-            const { item } = this.document;
-            if (item?.isOfType("spell") && item.system.area?.type === "burst") {
+            if (this.spellArea?.type === "burst") {
```
Comment (stwlam) [inline src/module/canvas/measured-template.ts:70]:
```suggestion
            if (this.effectArea?.type === "burst") {
```

File: src/module/item/spell/document.ts [hunk 2]
```diff
@@ -505,6 +505,7 @@ class SpellPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ite
                         traits: deepClone(this.system.traits.value),
                         ...this.getOriginData(),
                     },
+                    spellArea: this.system.area,
```
Comment (stwlam) [inline src/module/item/spell/document.ts:508]:
```suggestion
                    effectArea: this.system.area,
```

File: src/module/scene/measured-template-document.ts [hunk 3]
```diff
@@ -48,14 +60,17 @@ export class MeasuredTemplateDocumentPF2e<
...
-export interface MeasuredTemplateDocumentPF2e<TParent extends ScenePF2e | null = ScenePF2e | null>
+interface MeasuredTemplateDocumentPF2e<TParent extends ScenePF2e | null = ScenePF2e | null>
     extends MeasuredTemplateDocument<TParent> {
     get object(): MeasuredTemplatePF2e<this> | null;
 
     flags: DocumentFlags & {
         pf2e?: {
             messageId?: string;
             origin?: ItemOriginFlag;
+            spellArea?: Maybe<SpellArea>;
```
Comment (stwlam) [inline src/module/scene/measured-template-document.ts:71]:
```suggestion
            effectArea?: Maybe<SpellArea>;
```

File: src/module/scene/measured-template-document.ts [hunk 4]
```diff
@@ -31,6 +39,10 @@ export class MeasuredTemplateDocumentPF2e<
         return game.messages.get(this.flags.pf2e?.messageId ?? "") ?? null;
     }
 
+    get spellArea(): SpellArea | null {
+        return this.flags.pf2e?.spellArea ?? null;
+    }
```
Comment (stwlam) [inline src/module/scene/measured-template-document.ts:44]:
```suggestion
    get effectArea(): SpellArea | null {
        return this.flags.pf2e?.effectArea ?? null;
    }
```

File: src/module/canvas/measured-template.ts [hunk 5]
```diff
@@ -19,6 +21,22 @@ class MeasuredTemplatePF2e<
...
+        return this.document.item;
+    }
+
+    get message(): ChatMessagePF2e | null {
+        return this.document.message;
+    }
+
+    get spellArea(): SpellArea | null {
+        return this.document.spellArea;
+    }
```
Comment (stwlam) [inline src/module/canvas/measured-template.ts:38]:
```suggestion
    get effectArea(): SpellArea | null {
        return this.document.effectArea;
    }
```

**Comment by In3luki** [comment]
Updated

**Pull request merged**